crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1051,6 +1057,134 @@ impl PgManager {
             .await?;
         SuiSystemStateSummary::try_from(result)
     }
+
+    pub(crate) async fn fetch_staked_sui(
+        &self,
+        address: SuiAddress,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+    ) -> Result<Option<Connection<String, Stake>>, Error> {
+        let obj_filter = ObjectFilter {
+            package: None,
+            module: None,
+            ty: Some(MoveObjectType::staked_sui().to_string()),
+            owner: Some(address),
+            object_ids: None,
+            object_keys: None,
+        };
+        let system_state = self.fetch_latest_sui_system_state().await?;
+        let current_epoch_id = system_state.epoch_id;
+        // we want the object ids of staked sui
+        // The reason we're using this API and not get_staked_by_owner in the
+        // governance api is because that API is
```

Comment:
I don't think this comment still applies since we aren't using the governance api. I also prefer this approach much more, and with the StakedSui, we can call into get_delegated_stake instead of also relying on governance api to fetch the objects. This gives us greater control over how we fetch the objects to begin with